### PR TITLE
Improve taxon deletion workflow for taxons

### PR DESCRIPTION
**Pull request by Davidslv**
Trello: https://trello.com/c/ODbysu1S/178-content-tagger-improve-taxon-deletion-workflow-for-taxons

### Taxon with child taxons and tagged content
screen shot 2016-10-04 at 15 16 53

### Taxon with no child taxons or tagged content
screen shot 2016-10-03 at 11 19 21

### CODE REVIEW

File: app/controllers/taxons_controller.rb [hunk 1]
```diff
@@ -58,6 +58,14 @@ def update
     redirect_to taxons_path
   end
 
+  def deletion
```
Comment (tijmenb) [inline app/controllers/taxons_controller.rb]: Perhaps `confirm_delete`?

**Comment by tijmenb** [comment]
Can you add an integration test for the flow?

File: app/views/taxons/deletion.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    taxon: taxon,
+    children: children,
+    tagged: tagged } %>
+
+<% if children.any? %>
+  <%= render partial: 'children', locals: { children: children } %>
+<% end %>
+
+<% if tagged.any? %>
+  <%= render partial: 'tagged_content', locals: { tagged: tagged } %>
```
Comment (carvil) [inline app/views/taxons/deletion.html.erb]: We could also use this in [taxons#show](https://github.com/alphagov/content-tagger/blob/master/app/views/taxons/show.html.erb#L36-L56) and remove some code duplication, what do you think?

File: app/views/taxons/_deletion_warning.html.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<% if tagged.any? || children.any? %>
+<div class="alert alert-warning clearfix" role="warning">
+  <p>
+  This taxon has content tagged to it, deleting it will cause the content to not
```
Comment (carvil) [inline app/views/taxons/_deletion_warning.html.erb]: We could move this message to I18n, WDYT?

File: app/views/taxons/_deletion_warning.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  please confirm that you really want to delete this taxon.
+  </p>
+
+  <%= link_to taxon_path(taxon.content_id), method: :delete,
+    class: 'btn btn-md btn-danger pull-right left-space',
+    data: { confirm: I18n.t('messages.views.confirm') } do %>
+    <span class="glyphicon glyphicon-trash"></span>
+    <span>
+      <!-- <%= I18n.t('views.taxons.delete_taxon') %> -->
+      I understand and I want to delete it!
```
Comment (carvil) [inline app/views/taxons/_deletion_warning.html.erb]: Same here regarding I18n.

File: app/views/taxons/_deletion_warning.html.erb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    </span>
+  <% end %>
+
+  <%= link_to taxon_path(taxon.content_id), method: :delete,
+    class: 'btn btn-md btn-danger',
+    data: { confirm: I18n.t('messages.views.confirm') } do %>
+    <span class="glyphicon glyphicon-trash"></span>
+    <span>
+      <!-- <%= I18n.t('views.taxons.delete_taxon') %> -->
+      I understand and I want to delete it!
```
Comment (carvil) [inline app/views/taxons/_deletion_warning.html.erb]: Same here regarding I18n.

File: app/views/taxons/deletion.html.erb (hunk 2 above)
Comment (Davidslv) [inline app/views/taxons/deletion.html.erb]: Indeed, will do that

File: app/views/taxons/_deletion_warning.html.erb (hunk 3 above)
Comment (Davidslv) [inline app/views/taxons/_deletion_warning.html.erb]: I was thinking in having some approval on the wording first before moving it to a `I18n`, have to talk with someone to see what would be the correct wording here.

**Comment by Davidslv** [comment]
I've updated this PR, let me know if you spotted anything else @tijmenb @carvil

**Pull request merged**